TYPO3.Neos/Resources/Public/JavaScript/Content/Application.js

```diff
@@ -406,6 +406,12 @@ function(
 						$('title').html($htmlDom.filter('title').html());
 						$('link[rel="neos-site"]').attr('href', $htmlDom.filter('link[rel="neos-site"]').attr('href'));
 
+						var $pageStylesheets = $('link[rel="stylesheet"]:not(.neos-ui-asset)');
+						$pageStylesheets.remove();
+
+						var $loadedStylesheets = $htmlDom.filter('link[rel="stylesheet"]:not(.neos-ui-asset)');
+						$('head').append($loadedStylesheets);
```

Comment:
Well, we have to fully sync the existing stylesheets and the newly loaded stylesheets then. Because loading another page could as well remove other stylesheets.